### ISSUE TITLE
API docs: link to pub docs

### ISSUE THREAD

**Issue by chalin** [dart-lang/site-webdev#1440]
Docs are available via http://pub.dartlang.org/documentation

Stop generating our own dartdocs for angular packages.

---

- [x] Switch `site.api` from `https://www.dartdocs.org/documentation` to `https://pub.dartlang.org/documentation`. There should be very few uses, if any, of `site.api`. - #1549
- [ ] Double check that any direct use of `site.api` is a valid use.
- [ ] Change links to ACX API entries so that they point to gallery pages (#1533)
- [ ] For each package, link to the appropriate package's version's docs.
- ...

**Comment by kwalrath** [comment]
Before switching, we'll need to do a couple of things:

* Make sure having pub.dartlang.org be the canonical place is the best experience for our users.
  * It would be nice not to have to generate docs on our site. Faster build, simpler build, ...
  * We need to make sure people can find the docs they need, and that the docs are correct.
* Make sure the docs on pub.dartlang.org are generated correctly, and with no more libraries/APIs exposed than we need.
  * https://webdev.dartlang.org/api and https://webdev-dartlang-org-dev.firebaseapp.com/api have 6 angular* libraries in 5 groups (angular & angular.security, angular_components, angular_forms, angular_router, angular_test) and 4 Dart SDK libraries (dart:async, dart:convert, dart:core, dart:html). 
  * Many pub.dartlang.org doc sets expose many more libraries:
    * https://pub.dartlang.org/documentation/angular_components/0.8.0/ has dozens of libraries, two of which are named angular_2.
    * https://pub.dartlang.org/documentation/angular/5.0.0-alpha%2B8/ has 5 libraries (angular, angular.builder, angular.experimental, angular.security, di).
    * https://pub.dartlang.org/documentation/angular_router/2.0.0-alpha%2B8/ has 2 libraries (angular_router, testing).
    * https://pub.dartlang.org/documentation/angular_forms/2.0.0-alpha/ has just 1 library (which sounds right to me).
    * https://pub.dartlang.org/documentation/angular_test/2.0.0-alpha%2B6/ has 2 libraries (angular_test & angular_test.compatibility).

**Comment by chalin** [comment]
Just to be clear. I was thinking of keeping the current API landing page (see below), but redirect users to http://pub.dartlang.org/documentation for individual pages:

> <img width="800" alt="screen shot 2018-03-28 at 11 37 47" src="https://user-images.githubusercontent.com/4140793/38039976-7e19424c-327c-11e8-9f06-02ff7c0a2260.png">

**Comment by nshahan** [comment]
> https://pub.dartlang.org/documentation/angular_components/0.8.0/ has dozens of libraries, two of which are named angular_2.

What is the ask here? Are we saying that we don't want so many files exposed under lib to make the docs look better?

Linking a similar issue.
https://github.com/dart-lang/angular_components/issues/242

**Comment by kwalrath** [comment]
I think it's confusing to expose so many libraries in the docs, especially if the APIs are duplicated. People won't know which docs to look at. Maybe we could suppress the API docs for the small libraries, but mention in the relevant components how to include only that component? Or vice versa, if you really don't want people to import the uberlibrary.

**Comment by chalin** [comment]
@kevmoo, can we start using http://pub.dartlang.org/documentation now? If not, what are we waiting on?

**Comment by kevmoo** [comment]
> can we start using http://pub.dartlang.org/documentation now? If not, what are we waiting on?

I think so. CC @mit-mit @isoos @jakobr-google

**Comment by kwalrath** [comment]
Related: https://github.com/dart-lang/angular_components/issues/252

**Comment by mit-mit** [comment]
>  can we start using http://pub.dartlang.org/documentation now?

Yes, you can. That is where all pub docs are now hosted.

**Comment by chalin** [comment]
I've opened https://github.com/dart-lang/pub-dartlang-dart/issues/1216 as one means of addressing https://github.com/dart-lang/angular_components/issues/242 (which is also discussed here).

**Comment by chalin** [comment]
Superseded by https://github.com/dart-lang/site-angulardart/issues/70
